app/views_test.py

```diff
@@ -149,4 +149,168 @@ def test_unauthorized():
+            startTime = 23
+            endTime = 4
+            assert(views.validate_make_new_reservation_times(startTime, endTime).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+
+def test_valid_validate_make_new_reservation_date():
+    with app.app_context():
+        with app.test_request_context():
+            dateList = ["2040", "04", "04"]
+            assert(views.validate_make_new_reservation_date(dateList) is None)
+            dateList = ["2040", "4", "4"]
+            assert(views.validate_make_new_reservation_date(dateList) is None)
+
+def test_invalid_validate_make_new_reservation_date_more_than_3_elems():
+    with app.app_context():
+        with app.test_request_context():
+            dateList = ["2040", "04", "04", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+            dateList = ["2040", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+
+def test_invalid_validate_make_new_reservation_date_elem_is_not_digit():
+    with app.app_context():
+        with app.test_request_context():
+            dateList = ["2040", "04", "not a digit"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+            dateList = ["2040", "not a digit", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+            dateList = ["not a digit", "04", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+
+def test_invalid_validate_make_new_reservation_date_impossible_date():
+    with app.app_context():
+        with app.test_request_context():
+            dateList = ["2040", "04", "90"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+            dateList = ["2040", "49", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+
+def test_invalid_validate_make_new_reservation_date_before_current_date():
+    with app.app_context():
+        with app.test_request_context():
+            dateList = ["1999", "04", "04"]
+            assert(views.validate_make_new_reservation_date(dateList).status_code is views.STATUS_CODE['UNPROCESSABLE'])
+
+def test_valid_make_new_reservation_room_user_exists(monkeypatch):
+    with app.app_context():
+        with app.test_request_context():
+            roomId = "1"
+            userId = "5"
+            def mock_user_find(_):
+                return User(5, 'glorious', 'carpet')
+            monkeypatch.setattr(UserMapper, 'find', mock_user_find)
+            assert(views.validate_make_new_reservation_room_user_exists(roomId, userId) is None)
+
```

Comment:
Also, test when someone tries 

- To reserve a non existing room.
- To reserve a room that is already reserved
- To reserve a room with a valid date, but too far into the future (beyond a 4 month limit, or in the next semester?)
- Given a user that has a prior 2-hour reservation, that they attempt to make another 2-hour reservation (less than three hours, so it validates...), effectively going beyond their maximum of 3 hours a week. (Requirement 2 of new requirements) 
- QUESTION: Maybe it is not to be addressed in this PR, but how are we checking the requirement of not allowing users to make a repeated reservation for more than three weeks? If it should be addressed, then this scenario should be tested here...